### ISSUE TITLE
contr.helmert.weighted ordering issues

### ISSUE THREAD

**Issue by msonderegger** [sfraundorf/psycholing#3]
I think the implementation`contr.helmert.weighted` has unintended behavior when applied to a factor whose levels aren't in alphabetical order.

Example, using `regularity` from `languageR`:

```
# relevel to intuitive order
regularity$Auxiliary <- factor(regularity$Auxiliary, levels=c('hebben', 'zijnheb', 'zijn'))

# set contrasts to weighted helmert
contrasts(regularity$Auxiliary) <- contr.helmert.weighted(regularity$Auxiliary)
```

The resulting contrasts are:
```
> contrasts(regularity$Auxiliary) 
              [,1]       [,2]
hebben  -0.1021429 -0.1471429
zijnheb  0.3978571  0.8528571
zijn     0.8978571 -0.1471429
```
This would be the right contrast matrix if the row labels were `hebben`, `zijn`, `zijnheb` (alphabetical order). Compare what happens if you set contrasts with R's default Helmert:

```
> contrasts(regularity$Auxiliary) <- contr.helmert(3)
> contrasts(regularity$Auxiliary)
        [,1] [,2]
hebben    -1   -1
zijnheb    1   -1
zijn       0    2
```

Thanks for the `contr.helmert.weighted` and `contr.helmert.unweighted` functionality -- they were just what I was looking for!

**Comment by sfraundorf** [comment]
Belatedly fixed in commit b95d0bf

Thanks Morgan for providing such a good example of the issue!

**Comment by msonderegger** [comment]
Belatedly -- thanks for fixing!
